packages/mosip-api/src/routes/event-registration.ts

```diff
@@ -29,19 +29,16 @@ export const registrationEventHandler = async (
   request: OpenCRVSRequest,
   reply: FastifyReply,
 ) => {
+  try {
+    await request.jwtVerify();
+  } catch {
+    return reply.status(401).send({ error: "Unauthorized" });
+  }
+
```

Comment:
Do we want to verify the jwt on the route level? It's fine currently as there are only a couple of routes but it could get out of hand quickly. There is an example here on how to do that: https://github.com/fastify/fastify-jwt?tab=readme-ov-file#usage

Reply:
Amended. I'm not 100% sure how MOSIP will authenticate back to us when we subscribe to the events that they will be sending of the record progression, but lets amend after that. Easiest would be if they have a .well-known public key and they'd sign a JWT with that. Then we'd just allow JWT's from that pubkey as well.